hipscat/lsd2_io.py

```diff
@@ -0,0 +1,340 @@
+    
+    if source == 'local':
+        return hp.read_map(hips_fits_file)
+        
+    if source == 'azure':
+        #try:
+        fs = _get_azure_fs(hips_fits_file, _config)
+        azure_file = hips_fits_file.split('.dfs.core.windows.net/')[1]
+        with fs.open(azure_file) as f:
+            hdu = fits.open(f)
+            return hp.read_map(hdu)
+        #except:
+        #    raise Exception("An error occured with the adlfs download")
+            
+            
+def read_parquet(pathway, library='pandas', engine='pyarrow', hipsdir='catalog', columns=None, storage_options=None):
+    '''
+    lsd2 method to return parquet file from abstract file systems
+    utilizes pandas.read_parquet for single file pathways
+    or dask.from_map(pandas.read_parquet, ...) for multiple parquet files
+    
+    pathway: 
+      can be a string for a single file
+        -> returns a pandas.read_parquet() object
+      can be a string for the catalog pathway
+        -> returns a lazyily loaded dask dataframe object
+           from constructing a list of every parquet file
+           in the hipscat.
+      can be a list of parquet file pathways
+        -> returns a lazyily loaded dask dataframe object
+           from every parquet file in the list
+    
+    library:
+       pandas: for single parquet files
+       dask: for multiple parquet files
+    
+    engine:
+        only supporting pyarrow right now
+        
+    hipsdir:
+        can be 'catalog' or 'neighbor'
+    
+    columns:
+        list of columns to reduce the returned dataframe object
+    '''
+    if not library in ['pandas', 'dask']:
+        raise Exception(f"Invalid library '{library}', only 'pandas' and 'dask' are supported")
+    if not engine in ['pyarrow']:
+        raise Exception(f"Invalid engine: '{engine}', only 'pyarrow' is supported")
+    if not hipsdir in ['catalog', 'neighbor']:
+        raise Exception(f"Invalid hipsdir: '{hipsdir}', only 'catalog' and 'neighbor' are supported")
+    
+    if isinstance(pathway, list):
+        catalog_path = pathway[0].split(f"{hipsdir}/Norder")[0]
+    else:
+        catalog_path = pathway.split(f"{hipsdir}/Norder")[0]
+
+    catalog_path = catalog_path if catalog_path[-1] != '/' else catalog_path[:len(catalog_path)-1]
+
+    source, _config = _infer_file_source(catalog_path, storage_options)
+
+    parquet_meta_file = read_parquet_metadata(
+        os.path.join(catalog_path, hipsdir), storage_options
+    )
+    meta = []
+    reorder_columns = [] if columns is not None else None
+    for c in parquet_meta_file.pandas_metadata["columns"]:
+        if columns is not None and c['name'] != '_ID':
+            if c['name'] in columns:
+                meta.append((c["name"], c["pandas_type"]))
+                reorder_columns.append(c['name'])
+        elif c['name'] != '_ID':
+            meta.append((c["name"], c["pandas_type"]))
```

Comment:
I discovered last night that there is an easier (and probably better) way of generating the `meta` for Dask. Pyarrow parquet can read the schema from the metadata file and convert to an empty pandas DF from that. 
```
meta_file = pq.read_schema(path)
empty_table = meta_file.empty_table()
empty_df = empty_table.to_pandas()
```

Reply:
This solution actually worked seemlessly. I've implemented it